src/v/cluster/rm_stm.cc

```diff
@@ -1196,4 +1279,53 @@ ss::future<stm_snapshot> rm_stm::take_snapshot() {
     co_return stx_ss;
 }
 
+ss::future<> rm_stm::save_abort_snapshot(abort_snapshot snapshot) {
+    iobuf snapshot_data;
+    reflection::adl<abort_snapshot>{}.to(snapshot_data, snapshot);
+    auto snapshot_size = snapshot_data.size_bytes();
+
+    auto filename = fmt::format(
+      "abort.idx.{}.{}", snapshot.first, snapshot.last);
+    auto writer = co_await _abort_snapshot_mgr.start_snapshot(filename);
+
+    iobuf metadata_buf;
+    reflection::serialize(metadata_buf, abort_snapshot_version, snapshot_size);
```

Comment:
please use explicit concrete types for the serialization protocol. for example:

`auto snapshot_size = snapshot_data.size_bytes();`

would change the on-disk format if iobuf changed the type returned by size_bytes

Reply:
thanks, that's an awesome observation!